### ISSUE TITLE
On Unicode Input selection, <Enter> won't insert selected character 

### ISSUE THREAD

**Issue by Sdlion** [fcitx/fcitx#379]
Linux Mint 18.2 Cinnamon Release
fcitx 1:4.2.9.1-1ubuntu1.16.04.2

Steps to reproduce
1. Ctrl+Alt+Shift+u
2. Search a desired character. E.g. 0x2014 / Em dash
3. Press <tab> to select "em dash" and press <enter>.

Result: 
The <enter> character is passed into the text box/input, no character has been inserted, fcitx unicode addon state machine remains on search state with the text but disappears.

Expected result:
<enter> (or another *documented* key) works as a trigger to inserted into text box the selected character. It should work just as clicking the desired character.

**Comment by HuShifang** [comment]
Issue also applies to Kubuntu 17.10 64-bit with fcitx 4.2.9.1

**Comment by wengxt** [comment]
emm. ok. just FYI, by default you may use Alt + number key to select the candidate.

Adding enter key to select the first candidate seems to be a reasonable request.

**Comment by Sdlion** [comment]
Oh! That's the "A" on "A-1"?
I think it could be more clear as "Alt+1", at the time I couldn't find any reference about that shortcut.
BTW I think most people would expect `<enter>` or `<space>` to select whatever is selected after selecting with `<tab>`.
I'm sorry, using square brackets without escaping missed the original context of my post: select with `<tab>` and insert with `<enter>`.

**Comment by HuShifang** [comment]
Agreed, I thought that it was a reference to a grouping of options (i.e. the first screen would be "A1, A2, ..." and the second would be "B1, B2, ..."

Thanks for taking a look at it...

**Comment by ei14** [comment]
Sorry to necrobump, but what's going on here? I suddenly update Fcitx and, unlike before, <Enter> no longer inserts the selected character. It used to do this; why remove that feature? Also, Alt+number does not work. In, for example, the Chromium browser, Alt+number just switches the tab, causing the unicode menu to immediately close without entering the character.

**Comment by wengxt** [comment]
@ei14 I doubt it has anything to do with fcitx. relevant code haven't change in fcitx for years. latest release of fcitx4 is like 2 years ago.

It's most likely your im module is not properly configured or sth, or chromium did something werid, that cause chromium to not sending certain keys.

I just did a quick test and it seems to work here though.
